pkg/kubelet/rocket/rocket.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rocket
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/coreos/go-systemd/dbus"
+	"github.com/coreos/go-systemd/unit"
+	"github.com/golang/glog"
+)
+
+var (
+	tmpDirPath = "/tmp"
+)
+
+type RocketRuntime struct {
+	systemd  *dbus.Conn
+	endpoint string
+}
+
+func NewRocketRuntime(endpoint string) (*RocketRuntime, error) {
+	systemd, err := dbus.New()
+	if err != nil {
+		return nil, err
+	}
+	return &RocketRuntime{
+		systemd:  systemd,
+		endpoint: endpoint,
+	}, nil
+}
+
+func (r *RocketRuntime) runCommand(stdout bool, subCommand string, args ...string) ([]byte, error) {
+	glog.V(4).Info("run command:", subCommand, args)
+	var allArgs []string
+	allArgs = append(allArgs, subCommand)
+	allArgs = append(allArgs, args...)
+	cmd := exec.Command(r.endpoint, allArgs...)
+	if !stdout {
+		return nil, cmd.Start()
+	}
+	return cmd.Output()
+}
+
+func (r *RocketRuntime) constructRunCmd(uuid string) string {
+	return fmt.Sprintf("%s %s %s", r.endpoint, "run-prepared", uuid)
+}
+
+func (r *RocketRuntime) podToUnits(pod *api.BoundPod) ([]*unit.UnitOption, error) {
+	b, err := json.Marshal(pod)
+	if err != nil {
+		return nil, err
+	}
+
+	units := []*unit.UnitOption{
+		{
+			Section: "Unit",
+			Name:    "Description",
+			Value:   "k8s_rocket_pod",
+		},
+
+		{
+			Section: "Install",
+			Name:    "WantedBy",
+			Value:   "multi-user.target", // TODO(yifan): Hardcode.
+		},
+		{
+			Section: "X-K8S",
+			Name:    "POD",
+			Value:   string(b),
+		},
+	}
+	return units, nil
+}
+
+type PodStatus struct {
+	State string
+}
+
+func (r *RocketRuntime) getPodsStatus() (map[string]*PodStatus, error) {
+	status := make(map[string]*PodStatus)
+	output, err := r.runCommand(true, "list", "--no-legend")
+	glog.V(4).Infof("list output: %s", string(output))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(output) == 0 { // No containers running.
+		return nil, nil
+	}
+	for _, line := range strings.Split(strings.TrimSpace(string(output)), "\n") {
```

Comment:
Whenever you're parsing output like this it helps to document what the data should look like. Just dump an example of `rkt list --no-legend` somewhere around here.